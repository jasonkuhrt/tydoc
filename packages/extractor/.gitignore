node_modules
dist
scratch.ts
