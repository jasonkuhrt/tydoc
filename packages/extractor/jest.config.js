const process = require('process')

/** @typedef {import('ts-jest')} */
/** @type {import('@jest/types').Config.InitialOptions} */
module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'node',
  setupFiles: ['./test/setup.ts'],
  watchPlugins: [
    'jest-watch-suspend',
    'jest-watch-typeahead/filename',
    'jest-watch-typeahead/testname',
    [
      'jest-watch-toggle-config',
      {
        setting: 'verbose',
      },
    ],
    [
      'jest-watch-toggle-config',
      {
        setting: 'bail',
      },
    ],
  ],
  globals: {
    'ts-jest': {
      tsconfig: 'test/tsconfig.json',
      diagnostics: Boolean(process.env.CI),
    },
  },
  watchPathIgnorePatterns: ['<rootDir>/dist/', '<rootDir>/oclif.manifest.json'],
}
