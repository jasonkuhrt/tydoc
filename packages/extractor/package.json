{
  "name": "tydoc",
  "description": "The TypeScript documenter that meets you where you are",
  "version": "0.0.0-dripip",
  "main": "dist/index.js",
  "license": "MIT",
  "bin": "dist/cli/main.js",
  "files": [
    "dist",
    "types.d.ts",
    "types.js",
    "oclif.manifest.json"
  ],
  "scripts": {
    "build:docs:toc": "doctoc README.md",
    "build:docs:api": "ts-node scripts/build-api-docs",
    "build:docs": "yarn build:docs:api && yarn build:docs:toc",
    "build:oclif": "oclif-dev manifest",
    "build:ts": "tsc",
    "build": "yarn clean && yarn build:oclif && yarn build:ts",
    "dev": "tsc -w",
    "clean": "rm -rf dist tsconfig.tsbuildinfo",
    "dev:test": "jest --watch",
    "test": "jest",
    "test:dev": "yarn -s dev:test",
    "format": "prettier --write .",
    "prepublishOnly": "yarn build",
    "release:stable": "dripip stable",
    "release:preview": "dripip preview"
  },
  "dependencies": {
    "@microsoft/tsdoc": "^0.12.21",
    "@oclif/command": "^1.8.0",
    "@oclif/config": "^1.17.0",
    "@oclif/plugin-help": "^3.2.0",
    "@types/debug": "^4.1.5",
    "@types/lodash": "^4.14.165",
    "chaindown": "^0.1.0-next.1",
    "debug": "^4.3.1",
    "dedent": "^0.7.0",
    "doctoc": "^1.4.0",
    "fs-jetpack": "^4.1.0",
    "globby": "^11.0.1",
    "json5": "^2.1.3",
    "kleur": "^4.1.3",
    "lodash": "^4.17.20",
    "prettier": "^2.2.1",
    "ts-morph": "^9.1.0",
    "typescript": "^4.1.2"
  },
  "devDependencies": {
    "@oclif/dev-cli": "1.24.1",
    "@prisma-labs/prettier-config": "0.1.0",
    "@types/decompress": "^4",
    "@types/dedent": "^0.7.0",
    "@types/jest": "26.0.15",
    "@types/json5": "^0.0.30",
    "@types/node": "14.14.10",
    "@types/prettier": "2.1.5",
    "axios": "^0.21.0",
    "decompress": "^4.2.1",
    "dripip": "0.10.0",
    "jest": "26.6.3",
    "jest-validate": "26.6.2",
    "jest-watch-suspend": "1.1.2",
    "jest-watch-toggle-config": "2.0.1",
    "jest-watch-typeahead": "0.6.1",
    "tempy": "^1.0.0",
    "ts-jest": "26.4.4",
    "ts-node": "9.0.0",
    "ts-node-dev": "1.0.0",
    "type-fest": "^0.20.2"
  },
  "prettier": "@prisma-labs/prettier-config",
  "jest": {
    "preset": "ts-jest",
    "watchPlugins": [
      "jest-watch-suspend",
      "jest-watch-typeahead/filename",
      "jest-watch-typeahead/testname",
      [
        "jest-watch-toggle-config",
        {
          "setting": "verbose"
        }
      ],
      [
        "jest-watch-toggle-config",
        {
          "setting": "bail"
        }
      ]
    ],
    "snapshotSerializers": [
      "./test/__helpers__/snapshotSerializer.ts"
    ],
    "watchPathIgnorePatterns": [
      "<rootDir>/dist/",
      "<rootDir>/oclif.manifest.json"
    ],
    "setupFiles": [
      "./test/setup.ts"
    ],
    "globals": {
      "ts-jest": {
        "tsconfig": "test/tsconfig.json",
        "diagnostics": false
      }
    }
  },
  "oclif": {
    "commands": "./dist/cli/commands",
    "bin": "tydoc",
    "plugins": [
      "@oclif/plugin-help"
    ]
  }
}
