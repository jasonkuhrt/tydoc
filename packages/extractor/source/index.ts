export * from './extract'
export * as Reader from './reader'
