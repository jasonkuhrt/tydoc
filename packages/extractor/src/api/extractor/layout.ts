import createDebug from 'debug'
import * as dedent from 'dedent'
import * as kleur from 'kleur'
import * as path from 'path'
import * as tsm from 'ts-morph'
import { PackageJson } from 'type-fest'
import {
  absolutify,
  assertDirExists,
  assertFileExists,
  assertPathAbsolute,
  getPackageMain,
  readPackageJson,
  stripExtension,
} from '../lib/package-helpers'
import { applyDiagnosticFilters, DiagnosticFilter } from '../lib/ts-helpers'

const debug = createDebug('tydoc:package')

type Givens = {
  /**
   * Should the projectDir and sourceMainModulePath be validated that they exist on disk?
   *
   * @default true
   */
  validateExists?: boolean
  /**
   * On a TypeScript
   */
  validateTypeScriptDiagnostics?: boolean | DiagnosticFilter[]
  projectDir?: string
  sourceMainModulePath?: string
  packageJson?: PackageJson
  sourceDir?: string
  tsMorphProject?: tsm.Project
}

/**
 * Calculate and validate the project layout
 */
export function scan(givens?: Givens) {
  let projectDir: string

  /**
   * Find the project root
   */

  if (givens?.projectDir) {
    projectDir = givens.projectDir
    assertPathAbsolute(
      givens.projectDir,
      `Layout scan received a relative path for "projectDir". It must be an absolute path.`
    )
    if (givens.validateExists) {
      assertDirExists(
        givens.projectDir,
        'Layout received a path for "projectDir" that is not a directory or does not actually exist.'
      )
    }
    debug('prjDir set to %s -- taken from passed config ', projectDir)
  } else {
    projectDir = process.cwd()
    debug('prjDir set to %s -- taken from current working directory', projectDir)
  }

  /**
   * Read package.json
   */

  let packageMain: string

  const packageJson = givens?.packageJson ?? readPackageJson(projectDir)

  if (!packageJson) {
    throw new Error(`Your project at ${kleur.yellow(projectDir)} is missing a package.json file.`)
  }

  /**
   * Find the real package entrypoint
   */

  packageMain = path.join(projectDir, getPackageMain(packageJson))

  debug('packageMain is %s', packageMain)

  /**
   * Create a ts-morph project, including reading tsconfig.json
   */

  let tsMorphPoject

  if (givens?.tsMorphProject) {
    tsMorphPoject = givens.tsMorphProject
  } else {
    const tsConfigFilePath = path.join(projectDir, 'tsconfig.json')

    if (givens?.validateExists) {
      assertFileExists(
        tsConfigFilePath,
        `You must have a tsconfig.json file in the root of your project (${kleur.yellow(projectDir)}).`
      )
    }

    tsMorphPoject = new tsm.Project({ tsConfigFilePath })

    if (givens?.validateTypeScriptDiagnostics) {
      const diagnostics = tsMorphPoject.getPreEmitDiagnostics()
      if (diagnostics.length) {
        if (
          givens.validateTypeScriptDiagnostics === true ||
          applyDiagnosticFilters(givens.validateTypeScriptDiagnostics, diagnostics)
        ) {
          const message = tsMorphPoject.formatDiagnosticsWithColorAndContext(diagnostics)
          console.log(`
        Tydoc stopped extracting documentation becuase the package was found to have type errors. You should fix these and then try again. If you do not care about these type errors and want to try extracting documentation anyways then try using one of the following flags:\n  --ignore-diagnostics\n  --ignore-diagnostics-matching
      `)
          throw new Error(message)
        }
      }
    }
  }

  /**
   * Find the source dir
   */

  let sourceDir: string

  if (givens?.sourceDir) {
    sourceDir = absolutify(projectDir, givens.sourceDir)
    if (givens.validateExists) {
      assertDirExists(givens.sourceDir)
    }
  } else {
    // todo what is .rootDirs?
    const tsconfigRootDir = tsMorphPoject.compilerOptions.get().rootDir

    if (tsconfigRootDir) {
      if (path.isAbsolute(tsconfigRootDir)) {
        sourceDir = tsconfigRootDir
      } else {
        sourceDir = path.resolve(projectDir, tsconfigRootDir)
      }
    } else {
      sourceDir = projectDir
    }
  }

  debug('sourceDir set to %s', sourceDir)

  /**
   * Find the main _source_ module
   */

  let sourceMainModulePath: string

  if (givens?.sourceMainModulePath) {
    sourceMainModulePath = absolutify(projectDir, givens.sourceMainModulePath)
    if (givens.validateExists) {
      assertFileExists(
        sourceMainModulePath,
        'Layout received a path for "sourceMainModulePath" that is not a file or does not actually exist.'
      )
    }
    sourceMainModulePath = stripExtension(sourceMainModulePath)
  } else {
    let outDir: string

    const tsconfigOutDir = tsMorphPoject.compilerOptions.get().outDir

    if (tsconfigOutDir === undefined) {
      throw new Error(dedent`
        Your ${kleur.yellow('tsconfig.json')} compilerOptions does not have ${kleur.yellow(
        'compilerOptions.outDir'
      )} specified.
      
        Tydoc needs this information to discover the path to your source main entrypoint.

        You can pass the --sourceMainEntrypointPath flag to specify this path explicitly.

        Otherwise, please update your tsconfig.json so that it has an outDir set.
    `)
    }

    outDir = absolutify(projectDir, tsconfigOutDir)

    debug('outDir set to %s', outDir)

    /**
     * Default assumptions:
     *
     * - The package.json "main" will be pointed at/within tsconfig.json outDir. E.g. there is no post-tsc build steps.
     * - The module name pointed to by "main" will map directly to the main [TS] source file name. E.g. there is no bundle step.
     */

    const realMainModulePathRel = path.relative(outDir, packageMain)
    const realMainModuleDirRel = path.dirname(realMainModulePathRel)
    const mainModuleName = path.basename(realMainModulePathRel, '.js')
    const sourceMainModulePathRel = path.join(realMainModuleDirRel, mainModuleName)
    const sourceMainModulePathAbs = path.join(sourceDir, sourceMainModulePathRel)
    sourceMainModulePath = sourceMainModulePathAbs
  }

  debug('sourceMainModulePath is %s', sourceMainModulePath)

  /**
   * Get settings in package.json
   */

  let tydocSettingsInPackage

  if ((packageJson as any).tydoc) {
    tydocSettingsInPackage = (packageJson as any).tydoc
    debug('read user settings from package.json %O', tydocSettingsInPackage)
    // todo validate
  } else {
    debug('no user settings to read from package.json')
    tydocSettingsInPackage = {}
  }

  const layout = {
    projectDir,
    sourceDir,
    sourceMainModulePath,
    tydocSettingsInPackage,
    tsconfigPath: path.join(projectDir, 'tsconfig.json'),
    packageJson,
    tsMorphPoject,
  }

  debug(`layout calculated:`, layout)

  return layout
}
