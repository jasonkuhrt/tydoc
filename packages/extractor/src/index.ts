export * from './api/extractor/extract'
export { fromPublished } from './api/extractor/fromPublished'
export { Options as RenderMarkdownOptions, render as renderMarkdown } from './api/renderers/markdown'
