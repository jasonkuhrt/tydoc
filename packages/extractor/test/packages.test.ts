import { fromPublished } from '../src/api/extractor/extract'

test('can get EDD from "sponsorsme" package', async () => {
  const edd = await fromPublished({
    packageName: 'sponsorsme',
    packageVersion: '1.0.1',
  })
  expect(edd).toMatchSnapshot()
})

test('can get EDD from "execa" package', async () => {
  const edd = await fromPublished({
    packageName: 'execa',
    packageVersion: '4.1.0',
  })
  expect(edd).toMatchSnapshot()
})
