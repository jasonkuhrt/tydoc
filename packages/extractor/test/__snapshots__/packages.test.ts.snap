// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests extraction of packages sponsorsme 1`] = `
Object {
  modules: Array [
    Object {
      isMain: false,
      kind: module,
      location: Object {
        absoluteFilePath: /tmp/dir/package/dist/index.d.ts,
      },
      mainExport: Object {
        kind: object,
        props: Array [
          Object {
            kind: prop,
            name: token,
            type: Object {
              kind: primitive,
              type: any,
            },
          },
          Object {
            kind: prop,
            name: options,
            type: Object {
              kind: primitive,
              type: any,
            },
          },
          Object {
            kind: prop,
            name: cached,
            type: Object {
              kind: primitive,
              type: any,
            },
          },
          Object {
            kind: prop,
            name: cursor,
            type: Object {
              kind: primitive,
              type: any,
            },
          },
          Object {
            kind: prop,
            name: hasNextPage,
            type: Object {
              kind: primitive,
              type: any,
            },
          },
          Object {
            kind: prop,
            name: getInfo,
            type: Object {
              hasProps: false,
              isOverloaded: false,
              kind: callable,
              props: Array [],
              raw: Object {
                nodeFullText: /**
     * Returns information about the sponsor if it exists.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * await sponsors.getInfo("maticzav")
     * \`\`\`
     */
    getInfo(login: string): Promise<Optional<Sponsorship>>;,
                nodeText: getInfo(login: string): Promise<Optional<Sponsorship>>;,
                typeText: (login: string) => Promise<Sponsorship>,
              },
              sigs: Array [
                Object {
                  kind: sig,
                  params: Array [
                    Object {
                      kind: sigParam,
                      name: login,
                      type: Object {
                        kind: primitive,
                        type: string,
                      },
                    },
                  ],
                  return: Object {
                    kind: unsupported,
                    raw: Object {
                      nodeFullText: /**
 * Represents the completion of an asynchronous operation
 */
interface Promise<T> {
    /**
     * Attaches callbacks for the resolution and/or rejection of the Promise.
     * @param onfulfilled The callback to execute when the Promise is resolved.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of which ever callback is executed.
     */
    then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): Promise<TResult1 | TResult2>;

    /**
     * Attaches a callback for only the rejection of the Promise.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of the callback.
     */
    catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): Promise<T | TResult>;
},
                      nodeText: interface Promise<T> {
    /**
     * Attaches callbacks for the resolution and/or rejection of the Promise.
     * @param onfulfilled The callback to execute when the Promise is resolved.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of which ever callback is executed.
     */
    then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): Promise<TResult1 | TResult2>;

    /**
     * Attaches a callback for only the rejection of the Promise.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of the callback.
     */
    catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): Promise<T | TResult>;
},
                      typeText: Promise<Sponsorship>,
                    },
                  },
                },
              ],
            },
          },
          Object {
            kind: prop,
            name: isSponsor,
            type: Object {
              hasProps: false,
              isOverloaded: false,
              kind: callable,
              props: Array [],
              raw: Object {
                nodeFullText: /**
     * Tells whether there exists a sponsor with the given login.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * await sponsors.isSponsor("maticzav")
     * \`\`\`
     */
    isSponsor(login: string): Promise<boolean>;,
                nodeText: isSponsor(login: string): Promise<boolean>;,
                typeText: (login: string) => Promise<boolean>,
              },
              sigs: Array [
                Object {
                  kind: sig,
                  params: Array [
                    Object {
                      kind: sigParam,
                      name: login,
                      type: Object {
                        kind: primitive,
                        type: string,
                      },
                    },
                  ],
                  return: Object {
                    kind: unsupported,
                    raw: Object {
                      nodeFullText: /**
 * Represents the completion of an asynchronous operation
 */
interface Promise<T> {
    /**
     * Attaches callbacks for the resolution and/or rejection of the Promise.
     * @param onfulfilled The callback to execute when the Promise is resolved.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of which ever callback is executed.
     */
    then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): Promise<TResult1 | TResult2>;

    /**
     * Attaches a callback for only the rejection of the Promise.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of the callback.
     */
    catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): Promise<T | TResult>;
},
                      nodeText: interface Promise<T> {
    /**
     * Attaches callbacks for the resolution and/or rejection of the Promise.
     * @param onfulfilled The callback to execute when the Promise is resolved.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of which ever callback is executed.
     */
    then<TResult1 = T, TResult2 = never>(onfulfilled?: ((value: T) => TResult1 | PromiseLike<TResult1>) | undefined | null, onrejected?: ((reason: any) => TResult2 | PromiseLike<TResult2>) | undefined | null): Promise<TResult1 | TResult2>;

    /**
     * Attaches a callback for only the rejection of the Promise.
     * @param onrejected The callback to execute when the Promise is rejected.
     * @returns A Promise for the completion of the callback.
     */
    catch<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | undefined | null): Promise<T | TResult>;
},
                      typeText: Promise<boolean>,
                    },
                  },
                },
              ],
            },
          },
          Object {
            kind: prop,
            name: flush,
            type: Object {
              hasProps: false,
              isOverloaded: false,
              kind: callable,
              props: Array [],
              raw: Object {
                nodeFullText: /**
     * Clears the cache.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * sponsors.flush()
     * \`\`\`
     */
    flush(): void;,
                nodeText: flush(): void;,
                typeText: () => void,
              },
              sigs: Array [
                Object {
                  kind: sig,
                  params: Array [],
                  return: Object {
                    kind: primitive,
                    type: void,
                  },
                },
              ],
            },
          },
          Object {
            kind: prop,
            name: find,
            type: Object {
              kind: primitive,
              type: any,
            },
          },
          Object {
            kind: prop,
            name: sponsorshipFromEdge,
            type: Object {
              kind: primitive,
              type: any,
            },
          },
        ],
        raw: Object {
          nodeFullText: /**
 * # sponsorme
 *
 * A lightweight client that lets you check your GitHub sponsors.
 *
 * ## Installation
 *
 * \`\`\`bash
 * yarn add sponsorme
 * \`\`\`
 *
 * ## Example
 *
 * \`\`\`ts
 * let sponsors = Sponsors({ token: "token" })
 *
 * await sponsors.getInfo("maticzav")
 * \`\`\`
 *
 * That's it!
 */
/**
 * Creates a Sponsors client that you can use to get sponsors information.
 */
export default class Sponsors {
    private token;
    private options;
    private cached;
    private cursor;
    private hasNextPage;
    constructor(opts: SponsorsOptions);
    /**
     * Returns information about the sponsor if it exists.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * await sponsors.getInfo("maticzav")
     * \`\`\`
     */
    getInfo(login: string): Promise<Optional<Sponsorship>>;
    /**
     * Tells whether there exists a sponsor with the given login.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * await sponsors.isSponsor("maticzav")
     * \`\`\`
     */
    isSponsor(login: string): Promise<boolean>;
    /**
     * Clears the cache.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * sponsors.flush()
     * \`\`\`
     */
    flush(): void;
    /**
     * Internal helper that finds a sponsoring for a given login.
     */
    private find;
    /**
     * Converts response's edge to Sponsorship abstract type.
     */
    private sponsorshipFromEdge;
    private static query;
},
          nodeText: export default class Sponsors {
    private token;
    private options;
    private cached;
    private cursor;
    private hasNextPage;
    constructor(opts: SponsorsOptions);
    /**
     * Returns information about the sponsor if it exists.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * await sponsors.getInfo("maticzav")
     * \`\`\`
     */
    getInfo(login: string): Promise<Optional<Sponsorship>>;
    /**
     * Tells whether there exists a sponsor with the given login.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * await sponsors.isSponsor("maticzav")
     * \`\`\`
     */
    isSponsor(login: string): Promise<boolean>;
    /**
     * Clears the cache.
     *
     * \`\`\`ts
     * let sponsors = Sponsors({ token: "token" })
     * sponsors.flush()
     * \`\`\`
     */
    flush(): void;
    /**
     * Internal helper that finds a sponsoring for a given login.
     */
    private find;
    /**
     * Converts response's edge to Sponsorship abstract type.
     */
    private sponsorshipFromEdge;
    private static query;
},
          typeText: Sponsors,
        },
      },
      name: index,
      namedExports: Array [
        Object {
          isTerm: false,
          isType: true,
          kind: export,
          name: SponsorsOptions,
          type: Object {
            kind: typeIndexRef,
            link: (dist/index).SponsorsOptions,
          },
        },
        Object {
          isTerm: false,
          isType: true,
          kind: export,
          name: Sponsorship,
          type: Object {
            kind: typeIndexRef,
            link: (dist/index).Sponsorship,
          },
        },
        Object {
          isTerm: false,
          isType: true,
          kind: export,
          name: SponsorshipSponsor,
          type: Object {
            kind: typeIndexRef,
            link: (dist/index).SponsorshipSponsor,
          },
        },
        Object {
          isTerm: false,
          isType: true,
          kind: export,
          name: SponsorshipTier,
          type: Object {
            kind: typeIndexRef,
            link: (dist/index).SponsorshipTier,
          },
        },
      ],
      path: /dist/index,
      tsdoc: null,
    },
  ],
  typeIndex: Object {
    (dist/index).SponsorsOptions: Object {
      kind: alias,
      name: SponsorsOptions,
      raw: Object {
        nodeFullText: export declare type SponsorsOptions = {
    /**
     * Whether the client should use internal cache or not.
     */
    cache?: boolean;
    /**
     * Viewer's GitHub token used for authentication.
     */
    token: string;
};,
        nodeText: export declare type SponsorsOptions = {
    /**
     * Whether the client should use internal cache or not.
     */
    cache?: boolean;
    /**
     * Viewer's GitHub token used for authentication.
     */
    token: string;
};,
        typeText: SponsorsOptions,
      },
      tsdoc: null,
      type: Object {
        kind: object,
        props: Array [
          Object {
            kind: prop,
            name: cache,
            type: Object {
              kind: primitive,
              type: boolean,
            },
          },
          Object {
            kind: prop,
            name: token,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
        ],
        raw: Object {
          nodeFullText: export declare type SponsorsOptions = {
    /**
     * Whether the client should use internal cache or not.
     */
    cache?: boolean;
    /**
     * Viewer's GitHub token used for authentication.
     */
    token: string;
};,
          nodeText: export declare type SponsorsOptions = {
    /**
     * Whether the client should use internal cache or not.
     */
    cache?: boolean;
    /**
     * Viewer's GitHub token used for authentication.
     */
    token: string;
};,
          typeText: SponsorsOptions,
        },
      },
    },
    (dist/index).Sponsorship: Object {
      kind: alias,
      name: Sponsorship,
      raw: Object {
        nodeFullText: export declare type Sponsorship = {
    id: string;
    createdAt: string;
    /**
     * Sponsor information.
     */
    sponsor: SponsorshipSponsor;
    /**
     * Whether sponsorship is publicly visible.
     */
    public: boolean;
    /**
     * Information about the tier.
     */
    tier: SponsorshipTier;
};,
        nodeText: export declare type Sponsorship = {
    id: string;
    createdAt: string;
    /**
     * Sponsor information.
     */
    sponsor: SponsorshipSponsor;
    /**
     * Whether sponsorship is publicly visible.
     */
    public: boolean;
    /**
     * Information about the tier.
     */
    tier: SponsorshipTier;
};,
        typeText: Sponsorship,
      },
      tsdoc: null,
      type: Object {
        kind: object,
        props: Array [
          Object {
            kind: prop,
            name: id,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
          Object {
            kind: prop,
            name: createdAt,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
          Object {
            kind: prop,
            name: sponsor,
            type: Object {
              kind: typeIndexRef,
              link: (dist/index).SponsorshipSponsor,
            },
          },
          Object {
            kind: prop,
            name: public,
            type: Object {
              kind: primitive,
              type: boolean,
            },
          },
          Object {
            kind: prop,
            name: tier,
            type: Object {
              kind: typeIndexRef,
              link: (dist/index).SponsorshipTier,
            },
          },
        ],
        raw: Object {
          nodeFullText: export declare type Sponsorship = {
    id: string;
    createdAt: string;
    /**
     * Sponsor information.
     */
    sponsor: SponsorshipSponsor;
    /**
     * Whether sponsorship is publicly visible.
     */
    public: boolean;
    /**
     * Information about the tier.
     */
    tier: SponsorshipTier;
};,
          nodeText: export declare type Sponsorship = {
    id: string;
    createdAt: string;
    /**
     * Sponsor information.
     */
    sponsor: SponsorshipSponsor;
    /**
     * Whether sponsorship is publicly visible.
     */
    public: boolean;
    /**
     * Information about the tier.
     */
    tier: SponsorshipTier;
};,
          typeText: Sponsorship,
        },
      },
    },
    (dist/index).SponsorshipSponsor: Object {
      kind: alias,
      name: SponsorshipSponsor,
      raw: Object {
        nodeFullText: export declare type SponsorshipSponsor = {
    /**
     * Login handle of the sponsor.
     */
    login: string;
    /**
     * Sponsor's email if available.
     */
    email?: string;
};,
        nodeText: export declare type SponsorshipSponsor = {
    /**
     * Login handle of the sponsor.
     */
    login: string;
    /**
     * Sponsor's email if available.
     */
    email?: string;
};,
        typeText: SponsorshipSponsor,
      },
      tsdoc: null,
      type: Object {
        kind: object,
        props: Array [
          Object {
            kind: prop,
            name: login,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
          Object {
            kind: prop,
            name: email,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
        ],
        raw: Object {
          nodeFullText: export declare type SponsorshipSponsor = {
    /**
     * Login handle of the sponsor.
     */
    login: string;
    /**
     * Sponsor's email if available.
     */
    email?: string;
};,
          nodeText: export declare type SponsorshipSponsor = {
    /**
     * Login handle of the sponsor.
     */
    login: string;
    /**
     * Sponsor's email if available.
     */
    email?: string;
};,
          typeText: SponsorshipSponsor,
        },
      },
    },
    (dist/index).SponsorshipTier: Object {
      kind: alias,
      name: SponsorshipTier,
      raw: Object {
        nodeFullText: export declare type SponsorshipTier = {
    id: string;
    createdAt: string;
    /**
     * Information about the tier.
     */
    name: string;
    description: string;
    /**
     * The amount sponsoree receives each month.
     */
    monthlyPriceInCents: number;
};,
        nodeText: export declare type SponsorshipTier = {
    id: string;
    createdAt: string;
    /**
     * Information about the tier.
     */
    name: string;
    description: string;
    /**
     * The amount sponsoree receives each month.
     */
    monthlyPriceInCents: number;
};,
        typeText: SponsorshipTier,
      },
      tsdoc: null,
      type: Object {
        kind: object,
        props: Array [
          Object {
            kind: prop,
            name: id,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
          Object {
            kind: prop,
            name: createdAt,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
          Object {
            kind: prop,
            name: name,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
          Object {
            kind: prop,
            name: description,
            type: Object {
              kind: primitive,
              type: string,
            },
          },
          Object {
            kind: prop,
            name: monthlyPriceInCents,
            type: Object {
              kind: primitive,
              type: number,
            },
          },
        ],
        raw: Object {
          nodeFullText: export declare type SponsorshipTier = {
    id: string;
    createdAt: string;
    /**
     * Information about the tier.
     */
    name: string;
    description: string;
    /**
     * The amount sponsoree receives each month.
     */
    monthlyPriceInCents: number;
};,
          nodeText: export declare type SponsorshipTier = {
    id: string;
    createdAt: string;
    /**
     * Information about the tier.
     */
    name: string;
    description: string;
    /**
     * The amount sponsoree receives each month.
     */
    monthlyPriceInCents: number;
};,
          typeText: SponsorshipTier,
        },
      },
    },
  },
}
`;
